### DISCUSSION TITLE
[백엔드] 삼항 연산자 VS Optional VS 그냥 Getter

### DISCUSSION THREAD

**Discussion by lsh1215** [ReXume/techeer-resume#192] [General]
- Optional
![image](https://github.com/user-attachments/assets/de103e5b-0f67-4614-9313-d6c5a821857c)

- 삼항 연산자
![image](https://github.com/user-attachments/assets/2d85b26b-1b36-4b93-9cbb-a87021b6e270)

- 그냥 Getter
![image](https://github.com/user-attachments/assets/13737e9a-399c-4894-9389-37eb16b4bca7)

AI code review를  삼항 연산자 대신 Optional을 사용해서 코드를 간소화하고 가독성이 좋게 할 수 있다고 해서 코드 수정 과정에서 3가지 경우 모두 같은 동작을 하는 것을 확인

1. 그냥 getter를 쓰자 
   - 장점 : 코드가 굉장히 간결하고 가독성이 좋음
   - 단점 : null값이 나오는 지 코드를 보고 확인하기 힘듦

3. Optional이나 삼항 연산자를 쓰자
     - Optional이 가독성이 더 좋다
     - 삼항 연산자가 가독성이 더 좋다

이에 따라 선택이 필요할 듯

**Comment by YunJuwon0825** [comment]
삼항연산자가 어떤 값이 나오는지 확인하기 쉬워서 좋아!
조건 다음 id가 나올지 null이 나올지 순서대로 볼 수 있어서 바로 알 수 있는데
Getter는 가장 간결하지만 역시 null 값이 나오는지 안나온지 알 수 없어서 아쉬워.
Optional은 구조적으로는 삼항연산자와 같은데 orElse 또는 ofNullable 함수들이 있었어 보기 어려워.

**Comment by dlwldnjs1009** [comment]
https://velog.io/@tosspayments/null-%EB%A6%AC%ED%84%B4%EC%9D%80-%EC%99%9C-%EB%82%98%EC%81%A0%EA%B9%8C
위 블로그 참고 한 번 해봐!
